### PR TITLE
Add–1d–layers

### PR DESCRIPTION
**Pull request by dave-fernandes**
Added Conv1D, MaxPool1D and AvgPool1D layers

### CODE REVIEW

File: Tests/DeepLearningTests/LayerTests.swift [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+// limitations under the License.
+
+import XCTest
+@testable import DeepLearning
+
+final class LayerTests: XCTestCase {
+    func testConv1D() {
+        let filter = Tensor<Float>(ones: [3, 1, 2]) * Tensor<Float>([[[0.33333333, 1]]])
+        let bias = Tensor<Float>([0, 1])
+        let layer = Conv1D<Float>(filter: filter, bias: bias, activation: identity, stride: 1, padding: .valid)
```
Comment (rxwei) [inline Tests/DeepLearningTests/LayerTests.swift]: Fit within 100 columns.

**Comment by dave-fernandes** [comment]
The following PR must be merged before this one will compile:
https://github.com/apple/swift/pull/23355

File: Sources/DeepLearning/Layer.swift [hunk 2]
```diff
@@ -332,6 +332,125 @@ public extension Dense {
...
+    /// The padding algorithm for convolution.
+    @noDerivative public let padding: Padding
+
+    /// Creates a `Conv1D` layer with the specified filter, bias, activation function, stride, and
+    /// padding.
+    ///
+    /// - Parameters:
+    ///   - filter: The filter (width, inputChannels, outputChannels).
+    ///   - bias: The bias (dimensions: output channels).
+    ///   - activation: The activation activation.
```
Comment (dan-zheng) [inline Sources/DeepLearning/Layer.swift]:
```suggestion
    ///   - activation: The element-wise activation function.
```
Please also make this change elsewhere, including for `Conv2D`.

File: Sources/DeepLearning/Layer.swift [hunk 3]
```diff
@@ -332,6 +332,125 @@ public extension Dense {
...
+    /// The stride of the sliding window for temporal dimension.
+    @noDerivative public let stride: Int32
+    /// The padding algorithm for convolution.
+    @noDerivative public let padding: Padding
+
+    /// Creates a `Conv1D` layer with the specified filter, bias, activation function, stride, and
+    /// padding.
+    ///
+    /// - Parameters:
+    ///   - filter: The filter (width, inputChannels, outputChannels).
```
Comment (dan-zheng) [inline Sources/DeepLearning/Layer.swift]: For the `- Parameters:` doc comment of memberwise initializers, please copy the doc comments from each individual stored property.

For example:
```swift
/// - Parameters:
///   - filter: The 3-D convolution kernel.
///   - bias: The bias vector.
///   ...
```

Please also make this change elsewhere, including for `Conv2D`.

File: Sources/DeepLearning/Layer.swift [hunk 4]
```diff
@@ -332,6 +332,125 @@ public extension Dense {
...
+    }
+}
+
+public extension Conv1D {
+    /// Creates a `Conv1D` layer with the specified filter shape, strides, padding, and
+    /// element-wise activation function. The filter tensor is initialized using Glorot uniform
+    /// initialization with the specified seed. The bias vector is initialized with zeros.
+    ///
+    /// - Parameters:
+    ///   - filterShape: The shape of the filter (width, inputChannels, outputChannels).
```
Comment (dan-zheng) [inline Sources/DeepLearning/Layer.swift]: Please use `` `[dim0, dim1, ...]` `` syntax for specifying all shapes in doc comments.

```suggestion
    ///   - filterShape: The 3-D shape of the filter, representing
    ///     `[width, inputChannels, outputChannels]`.
```

File: Sources/DeepLearning/Layer.swift [hunk 5]
```diff
@@ -332,6 +332,125 @@ public extension Dense {
...
+        stride: Int = 1,
+        padding: Padding = .valid,
+        activation: @escaping Activation = identity,
+        seed: (Int64, Int64) = (Int64.random(in: Int64.min..<Int64.max),
+                                Int64.random(in: Int64.min..<Int64.max))
+    ) {
+        let filterTensorShape = TensorShape([
+            Int32(filterShape.0), Int32(filterShape.1), Int32(filterShape.2)])
+        self.init(
+          filter: Tensor(glorotUniform: filterTensorShape, seed: seed),
```
Comment (rxwei) [inline Sources/DeepLearning/Layer.swift]: Indent by 4 from `self`.

File: Sources/DeepLearning/Layer.swift [hunk 6]
```diff
@@ -332,6 +332,125 @@ public extension Dense {
...
+    /// - Parameters:
+    ///   - filter: The filter (width, inputChannels, outputChannels).
+    ///   - bias: The bias (dimensions: output channels).
+    ///   - activation: The activation activation.
+    ///   - stride: The stride.
+    ///   - padding: The padding.
+    public init(
+        filter: Tensor<Scalar>,
+        bias: Tensor<Scalar>,
+        activation: @escaping Activation,
```
[the remark this answers is not in the document]
Comment (dave-fernandes) [reply Sources/DeepLearning/Layer.swift]: The init method in the struct declaration doesn't have a default. Only the ones in the extensions have default values. So this is consistent, but maybe not what is desired?

**Comment by dave-fernandes** [comment]
Regarding the APIs, I think MaxPool2D and AvgPool2D should use 2-tuples instead of 4-tuples for strides and poolSize. The 4-tuples are inconsistent with Conv2D.

File: Sources/DeepLearning/Layer.swift (hunk 6 above)
[the remark this answers is not in the document]
Comment (rxwei) [reply Sources/DeepLearning/Layer.swift]: Oh, you are right!

File: Sources/DeepLearning/Layer.swift [hunk 7]
```diff
@@ -332,6 +332,127 @@ public extension Dense {
...
+    ) {
+        let filterTensorShape = TensorShape([
+            Int32(filterShape.0), Int32(filterShape.1), Int32(filterShape.2)])
+        self.init(
+            filter: Tensor(glorotUniform: filterTensorShape, seed: seed),
+            bias: Tensor(zeros: TensorShape([Int32(filterShape.2)])),
+            activation: activation,
+            stride: Int32(stride),
+            padding: padding)
+        }
```
Comment (rxwei) [inline Sources/DeepLearning/Layer.swift]:
```suggestion
    }
```

**Pull request merged**